l2geth/miner/worker.go

```diff
@@ -614,6 +616,15 @@ func (w *worker) batchAnswerLoop() {
 			} else {
 				w.knowBatchPeriodAnswerMsg.Add(ev.Msg.Hash())
 			}
+			if ev.Msg.BatchIndex != w.currentBps.BatchIndex {
```

Comment:
currentBps will be updated and read in different go routine. Please add lock to avoid concurrent issue.

Reply:
LGTM